src/EditorFeatures/Core/Implementation/Diagnostics/AbstractDiagnosticsTaggerProvider.AggregatingTagger.cs

```diff
@@ -198,6 +210,52 @@ private void OnDiagnosticsUpdated(DiagnosticsUpdatedArgs e)
                 }
             }
 
+            private void ProcessDocumentOrProjectRemoval(DiagnosticsUpdatedArgs e)
+            {
+                if (e.Solution != null)
```

Comment:
See also [this](http://source.roslyn.io/#Microsoft.CodeAnalysis.EditorFeatures/Implementation/Diagnostics/AbstractDiagnosticsTaggerProvider.AggregatingTagger.cs,144), which implies you might also want to look at Document?